### ISSUE TITLE
Technical question: necessary steps for adding genes

### ISSUE THREAD

**Issue by ajpar94** [PharmGKB/PharmCAT#108]
* **Do you want to request a *feature* or report a *bug*?**
Neither. I would like to ask a technical question.

If I wanted to use PharmCAT's Named Allele Matcher to determine the diplotype of a gene that is currently not supported by you, what would I have to do?

For example: The gene **NAT2** has a translation table on pharmgkb (https://api.pharmgkb.org/v1/download/submission/1447964753).
If I turned this table into a JSON file called `NAT2_translation.json` and placed it into [src/main/resources/org/pharmgkb/pharmcat/definition/alleles/](https://github.com/PharmGKB/PharmCAT/tree/development/src/main/resources/org/pharmgkb/pharmcat/definition/alleles) (provided it has the same format as the other files), built and ran PharmCAT, would that be enough so that I see the result in the output files: `<my_sample>.matcher.json` and `<my_sample>.matcher.html`?

What else would I need to do? Are you, by any chance, planning to incorporate NAT2 yourself?

Thank you in advance!

**Comment by markwoon** [comment]
Is your intention just to get a named allele call?  Or do you also intend to get a full PharmCAT report?

**Comment by ajpar94** [comment]
Thanks for your reply. The allele call would be enough. I believe I found the solution in the documentation: https://pharmcat.org/technical-docs/Running-PharmCAT#custom-data-1

> **Running the NamedAlleleMatcher** 
> ...
> * **-d** <definitions_dir> = optional, a directory containing allele definitions to use instead of the default packaged allele definitions

All I would need to do is create a `NAT2_translation.json`, correct?

**Comment by markwoon** [comment]
Yes, that is the theory.

The trick is not in the formatting of the translation file but it's content.  The challenge is translating the alleles used by the translation tables on PharmGKB into something that works in VCF.  I don't have the time to document the process right now, but the code that handles the work is in the `DataManager` class.  Once you have the translation file, you will also need to generate the positions.vcf file if you also want to use the PharmCAT preprocessor.

It's doable but non-trivial, and the only documentation is the code at the moment.

FYI, we plan on adding NAT2 in the future, but not until we have the FDA guidance for it.

**Comment by ajpar94** [comment]
I was able to create a `NAT2_translation.json` by looking up the variants in the translation table manually and noting the genomic position, ref and alt according to VCF standards. In addition, I created a new positions.vcf file as you said.
I have tried it with a few VCFs and got the expected diplotype call. So it seems to be working. I will nonetheless have a look at `DataManager` class.

Thank you for your help.
